Sources/ConfidenceProvider/Cache/InMemoryProviderCache.swift

```diff
@@ -0,0 +1,67 @@
+import Combine
+import Foundation
+import OpenFeature
+import os
+
+public class InMemoryProviderCache: ProviderCache {
+    private var rwCacheQueue = DispatchQueue(label: "com.confidence.cache.rw", attributes: .concurrent)
+    static let currentVersion = "0.0.1"
+    private let cache: [String: ResolvedValue]
+
+    private var storage: Storage
+    private var curResolveToken: String?
+    private var curEvalContextHash: String?
+
+    init(storage: Storage, cache: [String: ResolvedValue], curResolveToken: String?, curEvalContextHash: String?) {
+        self.storage = storage
+        self.cache = cache
+        self.curResolveToken = curResolveToken
+        self.curEvalContextHash = curEvalContextHash
+    }
+
+    public func getValue(flag: String, ctx: EvaluationContext) throws -> CacheGetValueResult? {
+        if let value = self.cache[flag] {
+            guard let curResolveToken = curResolveToken else {
```

Comment:
A recent swift change means you can omit the rhs of this: `guard let curResolveToken`

Reply:
wow nice thanks!